CIP-NFTMetadataStandard.md

```diff
@@ -0,0 +1,119 @@
+---
+CIP:
+Title: NFT Metadata Standard
+Authors: Alessandro Konrad <alessandro.konrad@live.de>, Smaug <smaug@pool.pm>
+Comments-URI:
+Status: Draft
+Type: Informational
+Created: 2021-04-08
+Post-History: https://forum.cardano.org/t/cip-nft-metadata-standard/45687 and https://www.reddit.com/r/CardanoDevelopers/comments/mkhlv8/nft_metadata_standard/
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This proposal defines an NFT Metadata Standard for Native Tokens.
+
+## Motivation
+
+Tokens on Cardano are a part of the ledger. Unlike on Ethereum, where metadata can be attached to a token through a smart contract, this isn't possible on Cardano because tokens are native and Cardano uses a UTxO ledger, which makes it hard to directly attach metadata to a token.
+So the link to the metadata needs to be established differently.
+Cardano has the ability to send metadata in a transaction, that's the way we can create a link between a token and the metadata. To make the link unique, the metadata should be appended to the same transaction, where the token forge happens:
+
+> Given a token in a EUTXOma ledger, we can ask “where did this token come from?” Since tokens
+> are always created in specific forging operations, we can always trace them back through their
+> transaction graph to their origin.
+
+(Section 4.1 in the paper: https://hydra.iohk.io/build/5400786/download/1/eutxoma.pdf)
+
+## Considerations
+
+That being said, we have unique metadata link to a token and can always prove that with 100% certainty. No one else can manipulate the link except if the policy allows it to ([update mechanism](#update-metadata-link-for-a-specific-token)).
+
+## Specification
+
+This is the registered `transaction_metadatum_label` value
+
+| transaction_metadatum_label | description  |
+| --------------------------- | ------------ |
+| 721                         | NFT Metadata |
+
+### Structure
+
+The structure allows for multiple token mints, also with different policies, in a single transaction.
+
+```
+{
+  "721": {
+    "<policy_id>": {
+      "<asset_name>": {
+        "name": "<name>",
+        "image": "<uri>",
```

Comment:
Important to note as Spacebudz utilized IPFS Cid v0 hashes and so the "ipfs:" prefix was acceptable. However, expanding the use case to IPFS Cid v1 hashes, the "ipfs:" prefix will cause the JSON value to exceed 64-characters and create unusable metadata format/structure. I would argue that something along the lines of the "CNFT" format with an "asset": {} descriptor is more future-proof.

Reply:
That's interesting. What about making the image field an array in case the uri exceeds 64 bytes. The concatenation of the strings is the uri. This basically allows for any lenght. In case the uri is less than 64 bytes you keep at as string without array. So it saves storage as well.

Reply:
Hrm, so the definition would be:
image: "typeof string|array"...

That would be potentially possible if it's not better to use a separate identifier like "url" to show the prefix, i.e.
url: "https://blockfrost.io/ipfs/",
image: "[ipfsHash]"
or
url: "https://ipfs.io/ipfs/",
image: "[ipfsHash]"

Reply:
Yes that makes total sense,although I would not prefer https gateways, when you use already ipfs. Better keep the ipfs:// protocol. One advantage tho of making it with an array is that you could also embed data URIs.

Reply:
This is a fair point, "url" could contain either "ipfs:", "https://path/to/thing/", or "data:" (probably other things I'm not thinking of as well). image may then contain string|array as necessary and it will just be up to the platform to handle gluing the pieces together in the case of an array.

Reply:
If in a metadata we associate multiple NFT tokens and try to mint 1 token to correspond to multiple NFTs will that work ? In principle this would be very tricky. It means that later ownership would be tied for all these NFT via this single token ? Can someone shed light on this ?

Reply:
If you mint the same token with quantity 100 you can easily split that up. They all refer to the same metadata and of course can have multiple owners.